vendor/README.md

```diff
@@ -0,0 +1,319 @@
+Vendor tests
+============
+
+When creating a badge for a new service or changing a badge's behavior, tests
+should be included. They serve three purposes:
+
+1. The contributor and revewer can easily verify the code works as
+   intended.
+
+2. When a badge stops working on the live server, maintainers can find out
+   right away.
+
+3. They speed up future contributors when they are debugging or improving a
+   badge.
+
+Contributors should take care to cover each part of a badge's functionality,
+and ideally, all code branches:
+
+- Typical case
+- Customized cases
+- Server errors
+- Not found errors
+- Parse errors
+
+
+Tutorial
+--------
+
+In this tutorial, we'll write tests for the Travis badge:
+
+```js
+camp.route(/^\/travis(-ci)?\/([^\/]+\/[^\/]+)(?:\/(.+))?\.(svg|png|gif|jpg|json)$/,
```

Comment:
In this example, I would like to discuss what needs testing and what not.
This can be done with the `// (1)`  annotations.